src/models/predicate/proven_v01.rs

```diff
@@ -0,0 +1,290 @@
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Builder {
+    pub(crate) id: TypeURI,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Completeness {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) arguments: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) environment: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) materials: Option<bool>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Recipe {
+    #[serde(rename = "type")]
+    pub(crate) typ: TypeURI,
+    #[serde(rename = "definedInMaterial")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) defined_in_material: Option<usize>,
+    #[serde(rename = "entryPoint")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) entry_point: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) arguments: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) environment: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Metadata {
+    #[serde(rename = "buildInvocationId")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) build_invocation_id: Option<String>,
+    #[serde(rename = "buildStartedOn")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) build_started_on: Option<TimeStamp>,
+    #[serde(rename = "buildFinishedOn")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) build_finished_on: Option<TimeStamp>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) completeness: Option<Completeness>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) reproducible: Option<bool>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Material {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) uri: Option<TypeURI>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) digest: Option<HashMap<String, String>>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+/// Predicate `ProvenV01` means the predicate of SLSA format.
+///
+/// [ProvenV01](https://slsa.dev/provenance/v0.1)
+/// can be used together with most states.
+pub struct ProvenV01 {
+    builder: Builder,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    recipe: Option<Recipe>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    metadata: Option<Metadata>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    materials: Option<Vec<Material>>,
+}
+
+impl From<LinkMetadata> for ProvenV01 {
```

Comment:
The right conversion here should be `From<MetsData>`, but the `MetaData` cannot be used to fetch any status for its design, so I use `LinkMetaData` for temporary resource, thus the test can be engaged.

Will be rewrite after the refactor of `MetaData`, relying #34